Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEditorInternal;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Microsoft.MixedReality.Toolkit.Build.Editor
+{
+    // TODO Include settings
+    //internal class ScriptReferenceRetargettingSettings : ScriptableObject
+    //{
+    //    public const string k_MyCustomSettingsPath = "Assets/Editor/MyCustomSettings.asset";
+
+    //    [SerializeField]
+    //    private int m_Number;
+
+    //    [SerializeField]
+    //    private string m_SomeString;
+
+    //    internal static ScriptReferenceRetargettingSettings GetOrCreateSettings()
+    //    {
+    //        var settings = AssetDatabase.LoadAssetAtPath<ScriptReferenceRetargettingSettings>(k_MyCustomSettingsPath);
+    //        if (settings == null)
+    //        {
+    //            settings = CreateInstance<ScriptReferenceRetargettingSettings>();
+    //            settings.m_Number = 42;
+    //            settings.m_SomeString = "The answer to the universe";
+    //            AssetDatabase.CreateAsset(settings, k_MyCustomSettingsPath);
+    //            AssetDatabase.SaveAssets();
+    //        }
+    //        return settings;
+    //    }
+
+    //    internal static SerializedObject GetSerializedSettings()
+    //    {
+    //        return new SerializedObject(GetOrCreateSettings());
+    //    }
+    //}
+
+    //internal static class ScriptReferenceRetargettingSettingsIMGUIRegister
+    //{
+    //    [SettingsProvider]
+    //    public static SettingsProvider CreateMyCustomSettingsProvider()
+    //    {
+    //        SettingsProvider provider = new SettingsProvider("Project/ScriptReferenceRetargetter", SettingsScope.Project)
+    //        {
+    //            // By default the last token of the path is used as display name if no label is provided.
+    //            label = "Script Reference Retargetter Settings",
+    //            // Create the SettingsProvider and initialize its drawing (IMGUI) function in place:
+    //            guiHandler = (searchContext) =>
+    //            {
+    //                var settings = ScriptReferenceRetargettingSettings.GetSerializedSettings();
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_Number"), new GUIContent("My Number"));
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_SomeString"), new GUIContent("My String"));
+    //            },
+
+    //            // Populate the search keywords to enable smart search filtering and label highlighting:
+    //            keywords = new HashSet<string>(new[] { "Number", "Some String" })
+    //        };
+
+    //        return provider;
+    //    }
+    //}
```

Comment:
Do you know what the plan is for uncommenting this code (i.e. is it better to: 1) delete this commented out code or 2) having the TODO reference a github issue with a description on how to reintroduce this code?)

Reply:
I will delete the code.

---
In reply to: [277821264](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277821264) [](ancestors = 277821264)